src/server/mod.rs

```diff
@@ -926,4 +380,572 @@ where
             });
         tokio::spawn(task);
     }
+
+    fn handle_with_auth(session: Arc<Mutex<Session<S, U>>>, next: impl Fn(Event) -> Result<Reply, FTPError>) -> impl Fn(Event) -> Result<Reply, FTPError> {
+        move |event| match event {
+            // internal messages and the below commands are except from auth checks.
+            Event::InternalMsg(_)
+            | Event::Command(Command::Help)
+            | Event::Command(Command::User { .. })
+            | Event::Command(Command::Pass { .. })
+            | Event::Command(Command::Auth { .. })
+            | Event::Command(Command::Feat)
+            | Event::Command(Command::Quit) => next(event),
+            _ => {
+                {
+                    let session = session.lock()?;
+                    if session.state != SessionState::WaitCmd {
+                        return Ok(Reply::new(ReplyCode::NotLoggedIn, "Please authenticate"));
+                    }
+                }
+                next(event)
+            }
+        }
+    }
+
+    fn handle_with_logging(next: impl Fn(Event) -> Result<Reply, FTPError>) -> impl Fn(Event) -> Result<Reply, FTPError> {
+        move |event| {
+            info!("Processing event {:?}", event);
```

Comment:
Am I right in concluding there's no way to disable this yet?

Reply:
The program using the lib controls the logging and can make all logging appear or not appear. In unFTP we use `-v` `-vv` etc for this. This still has the bug of logging the password but for me it was not part of the scope of this piece of work. This should now be really easy to add here.